src/avatar/Avatar.js

```diff
@@ -50,35 +56,33 @@ const Avatar = ({
   ...attributes
 }) => {
   const width =
-    typeof size === 'number'
-      ? size
-      : DEFAULT_SIZES[size] || DEFAULT_SIZES.small;
+    typeof size === 'number' ? size : avatarSizes[size] || avatarSizes.small;
   const height = width;
   const titleSize = width / 2;
   const iconSize = width / 2;
+
+  const editButton = {
+    ...defaultEditButton,
+    ...passedEditButton,
+  };
   const editButtonSize = editButton.size || (width + height) / 2 / 3;
 
   const Utils = showEditButton && (
     <TouchableHighlight
-      style={[
+      style={StyleSheet.flatten([
         styles.editButton,
         {
           width: editButtonSize,
           height: editButtonSize,
           borderRadius: editButtonSize / 2,
         },
         editButton.style,
-      ]}
+      ])}
       underlayColor={editButton.underlayColor}
       onPress={onEditPress}
     >
       <View>
-        <Icon
-          size={editButtonSize * 0.8}
-          name={editButton.iconName}
-          type={editButton.iconType}
-          color={editButton.iconColor}
-        />
+        <Icon size={editButtonSize * 0.8} {...editButton} />
```

Comment:
Shouldn't we use the `renderNode` helper ? This way people can easily pass a custom component for the edit button 🤙

```jsx
renderNode(Icon, passedEditButton, defaultEditButton) // Need to include the size in defaultEditButton though
```

Reply:
I guess that could work, although I don’t know is really using this feature since it wasn’t documented in docs😂. 

Should we change it now or wait on feedback(Someone requests it)?

Reply:
Actually let's postpone this and wait for feedbacks, that's not a big task anyway